File: core/src/main/java/eu/tsystems/mms/tic/testframework/internal/Timings.java
/*
 * Testerra
 *
 * (C) 2020,  Peter Lehmann, T-Systems Multimedia Solutions GmbH, Deutsche Telekom AG
 *
 * Deutsche Telekom AG and all other contributors /
 * copyright owners license this file to you under the Apache
 * License, Version 2.0 (the "License"); you may not use this
 * file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 */
package eu.tsystems.mms.tic.testframework.internal;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

public final class Timings {

    public static final long LARGE_LIMIT = 2000;

    public static final Map<Integer, Long> TIMING_GUIELEMENT_FIND = new ConcurrentHashMap<Integer, Long>();
    public static final Map<Integer, Long> TIMING_GUIELEMENT_FIND_WITH_PARENT = new ConcurrentHashMap<Integer, Long>();

    private static int findCounter = 0;

    private Timings() {

    }

    public static synchronized int raiseFindCounter() {
        findCounter++;
        return findCounter;
    }

    public static int getFindCounter() {
        return findCounter;
    }
}

File: report-ng/app/src/components/timings/timings.ts
/*
 * Testerra
 *
 * (C) 2023, Selina Natschke, Telekom MMS GmbH, Deutsche Telekom AG
 *
 * Deutsche Telekom AG and all other contributors /
 * copyright owners license this file to you under the Apache
 * License, Version 2.0 (the "License"); you may not use this
 * file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import {autoinject, PLATFORM} from 'aurelia-framework';
import {RouteConfig, Router, RouterConfiguration} from "aurelia-router";

@autoinject()
export class Timings {
    private _router: Router;

    configureRouter(config: RouterConfiguration, router: Router) {
        this._router = router;
        config.map([
            {
                route: '',
                moduleId: PLATFORM.moduleName('t-systems-aurelia-components/src/components/empty/empty'),
            },
            {
                route: 'tests',
                moduleId: PLATFORM.moduleName('./test-timings/test-timings'),
                nav: true,
                name: "tests",
                title: 'Tests',
                settings: {
                    icon: "center_focus_strong"
                }
            },
            {
                route: 'sessions',
                moduleId: PLATFORM.moduleName('./session-timings/session-timings'),
                nav: true,
                name: "sessions",
                title: 'Sessions',
                settings: {
                    icon: "list",
                }
            },
        ]);
    }

    activate(
        params: any,
        routeConfig: RouteConfig,
    ) {
        if (!routeConfig.hasChildRouter) {
            const navOptions = {replace: true};
            const enabledRouteConfig = this._router.routes.find(routeConfig => routeConfig.nav);
            this._router.navigateToRoute(enabledRouteConfig.name, {}, navOptions);
        }
    }

    private _tabClicked(routeConfig: RouteConfig) {
        if (this._router.currentInstruction.config.name !== routeConfig.name) {
            this._router.navigateToRoute(routeConfig.name);
        }
    }
}
